docs/contribute/developer.md

```diff
@@ -161,6 +160,14 @@ done
 
 ---
 
+## Making code changes
+
+Any time you change any of the files listed in the `deps` section in the `<binary name>_binary` `local_resource`,
+Tilt automatically rebuilds the go binary. As soon as the binary is rebuilt, Tilt pushes it (and only it) into the
+appropriate running container, and then restarts the process.
+
+---
+
```

Comment:
@LalatenduMohanty ^ See I added the only extra info/diff we had in the catalogd Tilt doc here.